### PR TITLE
Tests for table selection in Safari

### LINKED ISSUE #4287: Failing tests/plugins/tableselection/integrations/tabletools/tabletools tests
## Type of report

Failing test

## Provide detailed reproduction steps (if any)

1. Open http://tests.ckeditor.test:1030/tests/plugins/tableselection/integrations/tabletools/tabletools#tests%2Fplugins%2Ftableselection%2Fintegrations%2Ftabletools%2Ftabletools%20test%20delete%20nested%20cells%20(classic)

### Expected result

Test passes

### Actual result

This test and its inline counterpart fails in Safari 14:
Screenshot 2020-09-18 at 14 56 43

## Other details

* Browser: Safari 14
* Installed CKEditor plugins: tableselection, tabletools

### LINKED ISSUE #4306: Can't select last column if there is a header in the table
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open https://jsfiddle.net/Comandeer/2841j0cw/
2. Try to select the last column

### Expected result

The last column can be selected.

### Actual result

The last column can't be selected or the selection is selecting the whole table.

## Other details

* Browser: Safari 14
* OS: macOS
* CKEditor version: 4.7.0+
* Installed CKEditor plugins: tableselection

Behaviour of the bug is different in different CKE4 versions:
* **4.7.0–4.12.1**: the table selection is not possible for the last column, the native selection is kicking in;
* **4.13.0+**: the table selection is activated, but it selects the first row of the table or the whole table – it can't be dragged down the column.

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

Not needed

## What changes did you make?

I've ignored the unit test that fails in Safari and added a manual one to check if the upstream issue is still present.

## Which issues does your PR resolve?

Closes #4287.
Closes #4306.

### CODE REVIEW

**Review by jacekbogdanski — COMMENTED** [review]
Do the tests start failing from Safari 14? It seems to work when using Safari 13.1 and it's the latest version available on Browser Stack.

**Comment by Comandeer** [comment]
Yes, they fail only on Safari 14.

**Comment by jacekbogdanski** [comment]
Thanks, I'm removing myself from review as I don't have macOS to test it.

File: tests/plugins/tableselection/manual/safaritheadlastcolumn.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+				<td>Cell 5.3</td>
+				<td>Cell 5.4</td>
+			</tr>
+		</tbody>
+	</table>
+</div>
+
+<script src="../_helpers/tableselection.js"></script>
+<script>
+	if ( !CKEDITOR.env.safari ) {
```
Comment (Dumluregn) [inline tests/plugins/tableselection/manual/safaritheadlastcolumn.html:50]: As there is a part of test that requires mouse, we should either ignore the whole test on mobile or add a note that the second point shouldn't be executed there.

**Review by Dumluregn — CHANGES_REQUESTED** [review]
The manual test in the current form fails. I think it will be misguiding during the testing, so I'd propose to switch the `Expected` and `Unexpected` results so that test passed and additionally add the explicit note that if test is passing then the bug is still present.

The other solution is adding a note that the test should fail instead, but I'd rather go with a positive way as in the long term it should be clearer to understand.

Also manual test should be ignored on mobile (see the comment).

**Comment by Comandeer** [comment]
I've split the test into two: the one is for selecting the column with the button and the second one, ignored on mobile, to select with mouse.

**Review by Dumluregn — APPROVED** [review]
LGTM! 🎉

**Pull request merged**